File: 수학(구현)/백준_2004_조합 0의 개수/조합 0의 개수.py
def count_zero(n, k):
    cnt = 0
    i = k
    while i <= n:
        cnt += (n//i)
        i *= k
    return cnt

n, m = map(int, input().split(' '))

f = count_zero(n, 5) - count_zero(m, 5) - count_zero(n-m, 5)
t = count_zero(n, 2) - count_zero(m, 2) - count_zero(n-m, 2)

print(min(f, t))

File: 수학(구현)/백준_2004_조합 0의 개수/조합 0의 개수.cs
using System.Collections;
using static System.Console;
using System;

class Program
{
    static int count_zero(int n, int k)
    {
        int cnt = 0;

        for(long i = k; i <= n; i*= k)
        {
            cnt += (int)(n / i);
        }
        return cnt;
    }
    public static void Main()
    {
        int n, m;
        string[] s = ReadLine().Split(' ');
        n = int.Parse(s[0]);
        m = int.Parse(s[1]);

        int f = count_zero(n, 5) - count_zero(m, 5) - count_zero(n - m, 5);
        int t = count_zero(n, 2) - count_zero(m, 2) - count_zero(n - m, 2);
        Write(Math.Min(f, t));

    }
}